DAQ_System/Source/main.cpp

```diff
@@ -46,122 +48,57 @@ int main() {
     // File system declaration
     FATFileSystem   file_system("fs");
     
-    printf("--- Mbed OS filesystem example ---\n");
+    // buffer string used to write to a file
+    char buffer[512] = "\0"; 
+
+    // used for error checking when operating the block device
+    int status = 0;
+
+    // sensor values
+    int RPM = 0;
+    int tire_temp = 0;
+    int air_temp = 0;
+    int battery = 10;
+
+    printf("--- CSV File Test ---\n");
 
     // Try to mount the filesystem
     printf("Mounting the filesystem... ");
     fflush(stdout);
     Mount(&file_system, &block_device);
 
-    int status = 0;
     // Open the numbers file
-    printf("Opening \"/fs/numbers.txt\"... ");
+    printf("Opening \"/fs/data.csv\"... ");
     fflush(stdout);
 
-    FILE* numbers_file = FileOpen("/fs/numbers.txt");
-
-    char buffer[32] = "\0"; // buffer string used to write to a file
-
-    for (int i = 0; i < 10; i++) {
-        printf("\rWriting numbers (%d/%d)... ", i, 10);
-        fflush(stdout);
-        sprintf(buffer, "   %d\n", i); // we use this to print our desired format to the buffer string which is then written to the file
-        FileWrite(numbers_file, buffer);
+    FILE* data_file = FileOpen("/fs/data.csv");
+
+    // write the first row to the file with some arbitrary sensor names
+    FileWrite(data_file, "Time (sec), RPM, Tire Temp(F), Air Temp (F), Battery (V)");
+
+    // fill the file with arbitrary numbers (this is just a proof of concept, these are intentionally bs)
+    clock_t timer = clock();
+    for(int i = 0; i < 10; i++) {
+        snprintf(buffer, sizeof(buffer), "%f, %d, %d, %d, %d", ((float)timer)/CLOCKS_PER_SEC, RPM, tire_temp, air_temp, battery);
+        FileWrite(data_file, buffer);
+        RPM += 100;
+        tire_temp += 10;
+        air_temp++;
+        battery -= 10;
+        timer = clock() - timer;
```

Comment:
File writing worked. Just make sure to write a new line character to move on to the next csv row